src/core/tracking.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable no-underscore-dangle */
+
+import config from 'config';
+import log from 'core/logger';
+
+
+export class Tracking {
+
+  constructor({trackingId, trackingEnabled, _log = log} = {}) {
+    if (typeof window === 'undefined') {
+      return;
+    }
+    this._log = _log;
+    this.id = trackingId;
+    this.enabled = trackingEnabled && trackingId;
+    this.logPrefix = `[GA: ${trackingEnabled ? 'ON' : 'OFF'}]`;
+
+    if (this.enabled) {
+      _log.info('Tracking init (Analytics is ON)');
+      /* eslint-disable */
+      // Snippet from Google UA docs: http://bit.ly/1O6Dsdh
+      window.ga = window.ga || function() {(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
+      ga('create', trackingId, 'auto');
+      ga('send', 'pageview');
+      /* eslint-enable */
+    } else {
+      _log.info('Tracking init. (Analytics is OFF)');
+      if (!this.id) {
```

Comment:
There's a test that checks that "Missing tracking id" is logged when the id is missing, but I think this test would fail:

```js
  it('should not log missing id when disabled', () => {
    tracking = new Tracking({
      trackingId: undefined,
      trackingEnabled: false,
      _log: {
        info: sinon.stub(),
        warn: sinon.stub(),
      },
    });
    // eslint-disable-next-line no-underscore-dangle
    assert.ok(tracking._log.info.calledWith(sinon.match(/Analytics is OFF/)));
    // eslint-disable-next-line no-underscore-dangle
    assert.notOk(tracking._log.warn.called);
  });